frontend/src/components/order/orderCard.tsx

```diff
@@ -0,0 +1,72 @@
+import { OrderPublicWithProviders } from "@/client";
+import { StatusTag } from "@/components/common/statusTag.tsx";
+import { Box, Code, Flex, Table, Tag, Tbody, Td, Tr } from "@chakra-ui/react";
+
+interface Props {
+  order: OrderPublicWithProviders;
+}
+export function OrderCard({ order }: Props) {
+  return (
+    <Box p={6} boxShadow="md" borderRadius="md" bg="white">
+      <Flex justify="space-between" align="center" mb={4}>
+        <Box fontSize="lg" fontWeight="bold">
+          Order
+        </Box>
+        <Box>
+          Status <StatusTag status={order.status} />
+        </Box>
+      </Flex>
+
+      <Table size="sm" variant="simple">
+        <Tbody>
+          <Tr>
+            <Td fontWeight="bold">Providers</Td>
+            <Td>
+              {order.providers?.map(({ name }) => (
```

Comment:
If order.providers is undefined or an empty array, the .map() method could fail or display nothing.

```
<Td>
  {order.providers && order.providers.length > 0 ? (
    order.providers.map(({ name }) => (
      <Tag key={`${order.id}-${name}`} colorScheme="blue">
        {name}
      </Tag>
    ))
  ) : (
    <Tag colorScheme="gray">No Providers</Tag>
  )}
</Td>
```